### ISSUE TITLE
Recent gateway builds are failing on Travis in wsn transport, "Killed"

### ISSUE THREAD

**Issue by cmebarrow** [kaazing/gateway#423]
Most often the failure occurs while building wsn transport, as in builds 

- [1152](https://travis-ci.org/kaazing/gateway/builds/108111982)
- [1159](https://travis-ci.org/kaazing/gateway/builds/108409661)

The only clue that something went wrong is the word "Killed" in the build output, as in this extract from build 1159:
```
UTF8HandlingIT.sendTextMessageWithValidUTF8PayloadInOneFragment26 success
sendTextMessageWithValidUTF8PayloadInOneFragment27(org.kaazing.gateway.transport.wsn.autobahn.utf8handling.UTF8HandlingIT) starting
UTF8HandlingIT.sendTextMessageWithValidUTF8PayloadInOneFragment27 success
sendTextMessageWithValidUTF8PayloadInOneFragment28(org.kaazing.gateway.transport.wsn.autobahn.utf8handling.UTF8HandlingIT) starting
Killed
```

**Comment by cmebarrow** [comment]
According to [Common Build Problems](https://docs.travis-ci.com/user/common-build-problems/) question "My build script is killed without any error" this could be caused by exceeding 3GB of memory.

**Comment by cmebarrow** [comment]
I altered MethodExecutionTrace class in test.utils to printout memory stats if used memory exceeds 500MB. The following output file from [Travis build 1179](https://travis-ci.org/kaazing/gateway/builds/109934932) shows that excessive memory usage is the cause of the "Killed" failure:
[Travis.transport.wsn.fail.with.memory.stats.txt](https://files.zenhub.io/56c4d7ec5d3a5c4e4bb3cb8b)

**Comment by cmebarrow** [comment]
I further altered MethodExecutionTrace class in test.utils to call System.gc when used memory exceeds 500MB. This resolves the build failures, but a better fix would be find a way to set the java max heap size option (say -Xmx500MB)in the .travis.yml file.

**Comment by cmebarrow** [comment]
The above fix did not work. Trying again using the value -Xmx1024m

**Comment by cmebarrow** [comment]
It is likely that MAVEN_OPTS is not the way to go. According to the [Failsafe plugin documentation](http://maven.apache.org/surefire/maven-failsafe-plugin/integration-test-mojo.html) "JVM settings are not inherited from MAVEN_OPTS". I am trying out JAVA_OPTS environment variable.

**Comment by cmebarrow** [comment]
Issue is resolved by using the argLine failsafe configuration parameter in gateway pom.xml to set the -Xmx1024m java option. This change is included in PR #426.

**Comment by cmebarrow** [comment]
Resolved in PR https://github.com/kaazing/gateway/pull/426 by altering gateway pom.xml to include argLine failsafe parameter with -Xms1024m
